main/lsp/updates.cc

```diff
@@ -176,6 +177,13 @@ void LSPLoop::reIndexFromFileSystem() {
     }
 }
 
+void flattenAll(const core::GlobalState &gs, vector<ast::ParsedFile> &indexedCopies) {
```

Comment:
`flattenAll` mutates the ASTs, so this should consume a `vector<ast::ParsedFile>` and return a new `vector<ast::ParsedFile>`. As-is, I *think* you're moving away all of the old trees and throwing away the new ones.